source/read/count-documents.txt

```diff
@@ -0,0 +1,221 @@
+.. _java-rs-count:
+
+===============
+Count Documents
+===============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: number, amount, estimation, code example
+
+Overview
+---------
+
+In this guide, you can learn how to retrieve an accurate and estimated count of the
+number of documents in a collection.
+
+.. _java-rs-accurate-count:
+
+Retrieve an Accurate Count
+--------------------------
+
+Use the ``countDocuments()`` method to count the number of documents that are in a
+collection. To count the number of documents that match a specific search
+critera, pass a dictionary that includes a query filter to the
+``countDocuments()`` method.
+
+To learn more about specifying a query, see :ref:`java-rs-specify-query`.
+
+Count All Documents
+~~~~~~~~~~~~~~~~~~~
+
+To return a count of all documents in the collection, call the ``countDocuments()``
+method and pass in no parameters, as shown in the following example:
+
+.. code-block:: java
+
+   Publisher<Long> countPublisher = restaurants.countDocuments();
+   Flux.from(countPublisher).doOnNext(System.out::println).blockLast();
+
+Count Specific Documents
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To return a count of documents that match specific search criteria, specify your
+query in the ``countDocuments()`` method, as shown in the following example. To
+learn more about how to specify a query, see :ref:`java-rs-specify-query`.
+
+.. code-block:: java
+   
+   Publisher<Long> countPublisher = restaurants.countDocuments(
+           eq("cuisine", "Italian"));
+   Flux.from(countPublisher)
+           .doOnNext(System.out::println)
+           .blockLast();
+
+Customize Count Behavior
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``countDocuments()`` method by passing
+optional parameters to the method. The ``CountOptions`` class provides methods that modify
+the behavior of the ``countDocuments()`` method. To use the ``CountOptions``
+class, construct a new instance of the class, then call one or more of its methods
+to modify the count operation. You can chain these method calls together.
+
+To modify the behavior of the count operation, pass the class instance and
+chained method calls as the last argument to the ``countDocuments()`` method. 
+If you don't specify any options, the driver does not customize the count operation.
+
+The following table describes the options you can set to customize ``countDocuments()``:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+      
+   * - ``collation(Collation collation)``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the MongoDB Server manual.  
```

Comment:
Q: Is there a source constant for MongoDB Server? If so, good to use (applies throughout)